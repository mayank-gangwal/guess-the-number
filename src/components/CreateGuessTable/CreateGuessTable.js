import React from "react";

export default function CreateGuessTable(props) {
  return (
    <div id="CreateGuessTable">
      <hr />
      {props.guesses.map(row => (
        <div key={row.numberOfGuess}>
          Guess #{row.numberOfGuess}. You guessed number {row.guessedNumber} - {" "}
          {row.temperature}
        </div>
      ))}
    </div>
  );
}
