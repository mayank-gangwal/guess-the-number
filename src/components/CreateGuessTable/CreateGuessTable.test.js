import React from "react";
import { shallow } from "enzyme";
import CreateGuessTable from "./CreateGuessTable";
import { create } from "react-test-renderer";

describe("Examining the syntax of Jest tests", () => {
  const initialProp = [{
    numberOfGuess: 1,
    guessedNumber: 5,
    temperature: 'hot'
  }];
  const wrapper = shallow(<CreateGuessTable guesses={initialProp} />)

  it("should render without crashing", () => {
    shallow(<CreateGuessTable guesses={initialProp} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should show table text as expected", () => {
    const description = wrapper.find("div").at(1);
    expect(description.text()).toBe("Guess #1. You guessed number 5 -  hot");
  });
});
