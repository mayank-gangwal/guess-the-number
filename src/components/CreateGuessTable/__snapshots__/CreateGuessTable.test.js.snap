// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examining the syntax of Jest tests should match the snapshot 1`] = `ShallowWrapper {}`;
